python/ray/data/aggregate.py

```diff
@@ -71,17 +79,168 @@ def __init__(self):
         )
 
 
+def _null_init(init: Callable[[KeyType], AggType], k: KeyType):
```

Comment:
Can you pull the null handling out into a separate file and add a comment on the high level null strategy?